src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.Element.cs

```diff
@@ -111,29 +111,9 @@ public static JsonElement SerializeToElement(object? value, Type inputType, Json
         }
 
         private static JsonElement WriteElement<TValue>(in TValue value, JsonTypeInfo<TValue> jsonTypeInfo)
-        {
-            Debug.Assert(jsonTypeInfo.IsConfigured);
-            JsonSerializerOptions options = jsonTypeInfo.Options;
-
-            // For performance, share the same buffer across serialization and deserialization.
-            using var output = new PooledByteBufferWriter(options.DefaultBufferSize);
-            using var writer = new Utf8JsonWriter(output, options.GetWriterOptions());
-
-            WriteCore(writer, value, jsonTypeInfo);
-            return JsonElement.ParseValue(output.WrittenMemory.Span, options.GetDocumentOptions());
-        }
+            => WriteDocument(value, jsonTypeInfo).RootElement;
```

Comment:
By delegating `JsonElement`/`JsonDocument` serialization to the `WriteDocument`, this change introduces a subtle change to how buffers are handled: we no longer dispose the `PooledBufferWriter` and instead assign ownership of the rented buffer to the underlying `JsonDocument`. This approach has pros and cons:

The user doesn't have access to the underlying `JsonDocument` and therefore cannot return the buffers to the pool by disposing it. At the same time though, this avoids the buffer copying done by `JsonElement.ParseValue`.

@steveharter thoughts?

Reply:
> and therefore cannot return the buffers to the pool by disposing it

How/when do those buffers then find their way back for reuse?

Reply:
They never get returned, similar to a deserialized `JsonDocument` doesn't get disposed by the user.

Reply:
Wait... today, we rent an array from the pool, store it in a JsonDocument, and that array is never returned to the pool?  That's broken.  If for a success case we're not going to return the array, it shouldn't be using one from the pool.  Pooled arrays are valuable in that they're typically long-lived and have been promoted to gen2.

Reply:
In that case I'll revert that particular change and use the original copying approach.

Reply:
Please do... and if I understood correctly that we're already dropping pool arrays as part of JsonDocument on success paths, please open an issue about that as well.  Thanks.

Reply:
[Done](https://github.com/dotnet/runtime/pull/73338/commits/3604bcbe4f4cad74e2c430e1a45bb0a3180cbfc5) and [done](https://github.com/dotnet/runtime/issues/73398).